File: LeetCode/Simulation/Python/Leetcode2283-check-if-number-has-equal-digit-count-and-digit-value.py
class Solution:
    def digitCount(self, num: str) -> bool:
        mp = [0] * 10
        for i in num:
            mp[ord(i) - ord('0')] += 1
        for i in range(len(num)):
            cnt = int(num[i])
            if cnt != mp[i]:
                return False
        return True

File: LeetCode/Simulation/Rust/Leetcode2283-check-if-number-has-equal-digit-count-and-digit-value.rs
impl Solution {
    pub fn digit_count(num: String) -> bool {
        let mut mp = vec![0;10];
        for c in num.as_bytes() {
            mp[(c - b'0') as usize] += 1; 
        }
        for i in 0..num.len() {
            let cnt = (num.as_bytes()[i] - b'0') as usize;
            if mp[i] != cnt {
                return false;
            }
        }
        true
    }
}
